controllers/workspace/provision/object.go

```diff
@@ -91,9 +91,9 @@ func SyncObject(object runtime.Object, client client.Client, reqLogger logr.Logg
 		updateErr := client.Update(context.TODO(), object)
 		if errors.IsConflict(updateErr) {
 			fmt.Println("Suppressing conflict in ", objType.String())
-			return true, nil
+			return found, true, nil
 		}
-		return true, updateErr
+		return object, true, updateErr
```

Comment:
I agree, this could use some cleaning up. In general, we suppress conflict errors because they're semi-common and have no effect on functionality (the resolution is to just try again, since we have an out-of-date resource we're working on).

I'll push this fix to the follow-up PR for easier review.